### ISSUE TITLE
Add in-house dependency parser

### ISSUE THREAD

**Issue by christos-c** [IllinoisCogComp/illinois-cogcomp-nlp#178]
Since it will be helpful for many projects, it might a good idea to incorporate Mengxiong's dependency parser (based on @IllinoisCogComp/illinois-sl) to the main set of annotators -- that way we can get rid of the Stanford NLP dependency (at least if constituency parsing is not needed).

The latest code should be in [Mengxion's Gitlab repo](https://gitlab-beta.engr.illinois.edu/mliu60/illinois-depparse/tree/master), but maybe Mengxiong can point you to a more recent version if he's still working with the group.

Initial assignment is just for registering the issue, please re-assign accordingly.

**Comment by danr-ccg** [comment]
One comment on the dependency parser:

Yes, I think that we would like to incorporate it, but there are a few issues to resolve before we do, and we need to do it the right way.
The key issue, as far as I remember, was the features. We need to figure out the issues that Mengxiong identified with Edison, and figure out how to fix it, before we incorporate it.
He is still working, although we have not met since the beginning of the semester.

  Dan

**Comment by christos-c** [comment]
@mliu-dark-knight @danyaljj @mssammon I'm not sure what is the state of this project/issue, but @microth and I are collaborating to get his SRL system to work with our pipeline (including the dep. parser). 

If the latest code of @mliu-dark-knight's project is still in Gitlab, I have found and fixed several problems that were causing failures on the SRL end. I have issued a [PR here](https://gitlab-beta.engr.illinois.edu/mliu60/illinois-depparse/merge_requests/1) which at least when testing with a pre-trained model seems to work although the numbers are not good: 

```
%age correct undirected 0.8074647246244879
%age correct directed & unlabeled 0.7350933090578061
%age correct labeled 0.5184342284934
```

If there is something that is causing these low scores and can be easily fixed, I move to merge the Gitlab project here, since it will avoid future compatibility problems.

**Comment by christos-c** [comment]
@mssammon @danyaljj @mliu-dark-knight any updates on this?

**Comment by danyaljj** [comment]
I've messaged @mliu-dark-knight twice, but never got reply :man_facepalming:

**Comment by mssammon** [comment]
I spoke with @mliu-dark-knight toward the end of last semester.  He had the code working but needed to take some additional steps to integrate it into the pipeline.  Some international students have problems accessing email while away, which might explain his lack of response.

**Comment by danyaljj** [comment]
I emailed him.

**Comment by christos-c** [comment]
Thanks @mssammon and @danyaljj for following up on this. More pressing than the integration however, is the significantly low performance I got when running the Gitlab code. If @mliu-dark-knight responds I would urge him to see whether there is a bug in the code (or whether it's something I set up wrong in the configuration) by testing his code on a fresh machine.

**Comment by danyaljj** [comment]
@christos-c what repo are you using? I think he's developing under his own account (not the group repo)

**Comment by christos-c** [comment]
Yes, that's the repo I was using https://gitlab-beta.engr.illinois.edu/mliu60/illinois-depparse

**Comment by danyaljj** [comment]
Maybe we should fine another person to take over this project ...

**Comment by mssammon** [comment]
I suggest waiting until the new semester begins before giving up on him.

**Comment by microth** [comment]
@christos-c What model and data set are you using? As mentioned in my email (on Nov 22), the dependency parsing model that I sent you achieves a fairly okayish accuracy (~91% UAS/87% LAS). Those numbers were computed on the CoNLL 2009 data set, which is what I use for SRL. If you test the same model on Stanford/UD/other dependency relations, then of you course you will get lower numbers because those follow different definitions. I don't think it's fair to blame @mliu-dark-knight here, unless you are 100% sure that the low performance is related to his code.

**Comment by christos-c** [comment]
@microth I tested the model on the data in the project's directory which I assumed to be the standard CoNLL one. I also tried retraining the model with the same data and those were the numbers I got. And of course I'm not blaming @mliu-dark-knight for the low scores. In fact I'm not blaming anyone, just trying to understand what's happening (this could well be something I'm doing wrong on my end). I would only like more responsiveness on Mengxiong's part since this is a time-critical issue.

**Comment by microth** [comment]
@christos-c Okay, maybe I should have phrased my comment better. Anyway, I am not sure which project directory you are referring to, but if you have retrained/tested a model on your own and got such low numbers, then sure, something must have gone wrong. Have you tried using gold POS? Maybe the issue is somehow related to the outdated dependencies in his project. I got the numbers above after running the POS tagger from his code on the CoNLL training and test sets.

**Comment by microth** [comment]
P.S.: Oh, and yes, I completely agree about the responsiveness. Last time I heard from @mliu-dark-knight was on Dec 22. :(

**Comment by christos-c** [comment]
I tried using the default setup that existed on the main class. Anyway, there's now another problem, since @mliu-dark-knight didn't bother merging my [Merge Request](https://gitlab-beta.engr.illinois.edu/mliu60/illinois-depparse/merge_requests/1) (which was opened 3 weeks ago), but instead made changes to his version, leaving me with a conflict resolution I don't want to deal with. If it's possible, @danyaljj or @mssammon, can you check out [my fork](https://gitlab-beta.engr.illinois.edu/christod/illinois-depparse) and see if you get the same poor results? @mliu-dark-knight's recent changes (that cause the conflict) have basically imported the latest version of the illinois-cogcomp software, which is what I've also done anyway.

**Comment by danyaljj** [comment]
Sorry I can't do that at the moment. Swamped with other stuff 😭

**Comment by mssammon** [comment]
I tried to check out your fork but got this error:
mssammon@bronte:/shared/experiments/mssammon/workspace-github/christos$ git clone https://gitlab-beta.engr.illinois.edu/christod/illinois-depparse
Cloning into 'illinois-depparse'...
remote: Counting objects: 3931, done.
remote: Compressing objects: 100% (2694/2694), done.
remote: fatal: unable to read 3de3c0c43d91f044190a06ea77024f58a6839988
remote: aborting due to possible repository corruption on the remote side.
fatal: early EOFs:   3% (118/3931)   
fatal: index-pack failed

**Comment by danyaljj** [comment]
Sigh ... in these situations you can just download the zip file and extract it on your machine.

**Comment by mssammon** [comment]
good point :-)

**Comment by mssammon** [comment]
at least, it *would* be a good point if the download didn't have exactly the same corruption issue. :+1:

**Comment by microth** [comment]
Same problem here...

> unzip illinois-depparse-master-10f066c6b7ded83dc56acebe1e2c925278a517bc.zip 
Archive:  illinois-depparse-master-10f066c6b7ded83dc56acebe1e2c925278a517bc.zip
  End-of-central-directory signature not found.

> tar -xvf illinois-depparse-master-10f066c6b7ded83dc56acebe1e2c925278a517bc.tar
(...)
illinois-depparse-master-10f066c6b7ded83dc56acebe1e2c925278a517bc/src/main/java/edu/illinois/cs/cogcomp/depparse/LabeledEdisonFeatureGenerator.java
tar: Unexpected EOF in archive
tar: Unexpected EOF in archive
tar: Error is not recoverable: exiting now

**Comment by christos-c** [comment]
@danyaljj @mssammon @microth sorry about that, something must have gone during my attempt at resolving the conflicts. I've reset the head of my remote to the previous stable version, so cloning/downloading works now.
